src/main/java/com/vozniuk/bookstore/service/impl/OrderServiceImpl.java

```diff
@@ -25,27 +24,30 @@
 import lombok.RequiredArgsConstructor;
 import org.springframework.data.domain.Pageable;
 import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
 
 @Service
 @RequiredArgsConstructor
 public class OrderServiceImpl implements OrderService {
     private final OrderRepository orderRepository;
     private final OrderItemRepository orderItemRepository;
-    private final ShoppingCartRepository shoppingCartRepository;
-    private final AuthenticationService authenticationService;
+    private final ShoppingCartService shoppingCartService;
     private final OrderMapper orderMapper;
     private final OrderItemMapper orderItemMapper;
-    private final UserRepository userRepository;
+    private final UserService userService;
 
+    @Transactional
     @Override
     public OrderResponseDto save(OrderRequestDto requestDto) {
         Order order = new Order();
         order.setShippingAddress(requestDto.getShippingAddress());
-        order.setOrderDate(LocalDateTime.now());
-        ShoppingCart shoppingCart = shoppingCartRepository
-                .findByUserId(authenticationService.getUserId())
-                .orElseThrow(() -> new EntityNotFoundException("Can't find shopping cart to place "
-                        + "an order by user with id: " + authenticationService.getUserId()));
+        ShoppingCart shoppingCart = shoppingCartService.getCart();
+        getOrderFromShoppingCart(order, shoppingCart);
+        orderRepository.save(order);
+        return orderMapper.toDto(order);
+    }
+
+    private void getOrderFromShoppingCart(Order order, ShoppingCart shoppingCart) {
```

Comment:
word get associated with getting (receiving) something. Method getOrderFromShoppingCart should not return void